BibBuddy/app/src/main/java/de/bibbuddy/BookOnlineFragment.java

```diff
@@ -113,14 +115,17 @@ private void handleIsbnInput(int keyCode, KeyEvent event) {
     }
   }
 
-  private void handleAddBook(Book book) {
-    Toast.makeText(getActivity(), getString(R.string.added_book),
-                   Toast.LENGTH_LONG).show();
+  private void handleAddBook(Book book, List<Author> authors) {
+    try {
+      BookDao bookDao = new BookDao(new DatabaseHelper(getContext()));
+      bookDao.create(book, authors, shelfId);
```

Comment:
Why don't you use the BookModel here.

Reply:
Originally suggested by @claudia3 this way - will alter to BookModel here!

Reply:
Using BookModel and then the method "addBook" does not work. Reason: In addBook, the new book is appended to the "bookList" in the BookModel. However, bookList is never declared if we only use addBook(). Instead we would have to read the whole shelf out of the database first. This would then double the work that was already done inside the library...